src/misc/messages.cpp

```diff
@@ -21,227 +21,751 @@
 
 #include "dosbox.h"
 
-#include <clocale>
-#include <cstdio>
-#include <cstdlib>
-#include <cstring>
-#include <map>
-#include <vector>
-
-#include "std_filesystem.h"
-#include <string>
-#include <unordered_map>
-
+#include "../dos/dos_locale.h"
 #include "ansi_code_markup.h"
 #include "checks.h"
 #include "control.h"
 #include "cross.h"
 #include "fs_utils.h"
 #include "host_locale.h"
 #include "setup.h"
+#include "std_filesystem.h"
 #include "string_utils.h"
 #include "support.h"
 #include "unicode.h"
 
-#define LINE_IN_MAXLEN 2048
+#include <cctype>
+#include <fstream>
+#include <map>
+#include <set>
+#include <string>
 
 CHECK_NARROWING();
 
-static const char *msg_not_found = "Message not Found!\n";
+static const std::string MsgNotFound = " MESSAGE NOT FOUND! ";
+static const std::string MsgNotValid = " MESSAGE NOT VALID! ";
+
+// ***************************************************************************
+// Single message storage class
+// ***************************************************************************
 
 class Message {
-private:
-	std::string markup_msg = {};
-	std::string rendered_msg = {};
+public:
+	// Note: any message needs to be verified before it can be safely used!
 
-	std::map<uint16_t, std::string> rendered_msg_by_codepage = {};
+	// Constructor for original, English strings
+	Message(const MsgFlags flags, const std::string& text);
+	// Constructor for translated strings from external file
+	Message(const std::string& text);
 
-	static std::string to_dos(const std::string& in_str, const uint16_t code_page)
+	const std::string& Get(const bool raw_requested);
```

Comment:
Boolean flags are a bit of an anti-patterns (unless you use them as enabled/disabled flags in trivial cases).

Consider adding `GetRaw()` instead.

Reply:
OK, changed.